### PR TITLE
update to Visual Studio 2017 and MozillaBuild 3.0 on Windows

### PR DESCRIPTION
**Pull request by ctaggart**
This is the minimal changes to update to Visual Studio 2017 and MozillaBuild 3.0 on Windows.

Any chance someone can upload
https://ftp.mozilla.org/pub/mozilla/libraries/win32/MozillaBuildSetup-3.0.exe
to your Amazon S3 bucket?
http://servo-rust.s3.amazonaws.com/build/MozillaBuildSetup-3.0.exe

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/368)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -37,15 +38,17 @@ environment:
...
+  - ps: Start-FileDownload "https://ftp.mozilla.org/pub/mozilla/libraries/win32/MozillaBuildSetup-3.0.exe"
+  - ps: .\MozillaBuildSetup-3.0.exe /S | Out-Null
   - ps: Start-FileDownload "https://static.rust-lang.org/dist/rust-${env:TARGET}.exe" -FileName "rust-install.exe"
   - ps: .\rust-install.exe /VERYSILENT /NORESTART /DIR="C:\rust" | Out-Null
   - ps: $env:PATH="$env:PATH;C:\rust\bin"
 # I can't make this work with a multi-line script (e.g. via |+ etc.), so we copy the if before every command
   - if %TARGET:*-msvc=msvc%==msvc set BUILD_ENV=msvc
   - if %TARGET:*-gnu=gnu%==gnu set BUILD_ENV=gnu
-  - if %BUILD_ENV%==msvc call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\vcvars64.bat"
+  # - if %BUILD_ENV%==msvc call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\vcvars64.bat"
```
Comment (ctaggart) [inline appveyor.yml]: I can drop this comment.

File: appveyor.yml [hunk 2]
```diff
@@ -37,15 +38,17 @@ environment:
 
 # called after cloning, before building
 install:
-  - ps: Start-FileDownload "http://servo-rust.s3.amazonaws.com/build/MozillaBuildSetup-2.2.0.exe"
-  - ps: .\MozillaBuildSetup-2.2.0.exe /S | Out-Null
+  # - ps: Start-FileDownload "http://servo-rust.s3.amazonaws.com/build/MozillaBuildSetup-2.2.0.exe"
+  - ps: Start-FileDownload "https://ftp.mozilla.org/pub/mozilla/libraries/win32/MozillaBuildSetup-3.0.exe"
```
Comment (ctaggart) [inline appveyor.yml]: I can switch the URL if someone upload the exe to S3.

**Comment by jdm** [comment]
@fitzgen Maybe these changes will help the upgrade...

**Comment by jdm** [comment]
@edunham Are you able to upload stuff to the s3 bucket?

**Pull request closed without merging**